### DISCUSSION TITLE
Can CGNS be used to read Common Grid File format meshes? (.cgd)

### DISCUSSION THREAD

**Discussion by wandadars** [CGNS/CGNS#755] [CGNStalk]
I was looking at the NPARC archive for Wind-US cases and they have their meshes in a .cgd file format. I can't for the life of my find anything regarding this format or if it can be read by any known open source tools like CGNS.  Has anyone worked with this grid format before?

**Comment by brtnfld** [comment]
I've never worked with cgd, but here's a breakdown of what a Common Grid File is:

    Purpose: Stores data for multizone structured grids. These grids can also include associated boundary condition data.
    Structure: Common Grid Files use a flexible binary format for direct access.
    Software: This format is primarily used by the Wind-US Navier-Stokes flow solver and the CFPOST post-processing utility, both developed by NASA.

As for  it roots -- common file versions:
https://www.grc.nasa.gov/www/winddocs/cff/versions.html

I can't definitively say whether CGNS can read CGD files. However, it's likely not possible since  (v3) is based on CGNS but not necessarily compatible with it. Others may know for sure.

https://www.grc.nasa.gov/www/winddocs/user/user.pdf :

2.2.2 Creating the Common Grid (.cgd ) File
The computational grid used by Wind-US for a particular case is stored in a Common Grid
(.cgd ) file, so named because the file is formatted according to Boeing’s Common File Format.2 The
2Wind-US also supports the use of CGNS files for the grid and flow solution, using the CGNSBASE keyword. This
tutorial, however, uses common files

**Comment by Bob-Bush** [comment]
Cgd and cfl files are built on the adf core.  So, Cgnsview can open and then you can see the structure and contents.  But the cgns libs don't know how to interpret the contents, and the cgns software can't do a conversion.  There may be an old conversion program in the cgns archives.  Nparc has libs to read and write from fortran or c that would be needed to work with the files.
